### ISSUE TITLE
[Apple Intelligence] errors out

### ISSUE THREAD

**Issue by shreyshah33** [raycast/extensions#15337]
### Extension

https://www.raycast.com/EvanZhouDev/raycast-apple-intelligence

### Raycast Version

1.85.1

### macOS Version

15.1

### Description

<img width="258" alt="image" src="https://github.com/user-attachments/assets/0b07e07a-628e-4cf1-a0dc-d165e58fc8b3">
apple intelligence errors out

### Steps To Reproduce

Select text in Slack and then run the commands

### Current Behaviour

Errors

### Expected Behaviour

No errors

**Comment by ozalisky** [comment]
Hey @EvanZhouDev, here's the video for the issue.

Proofread works fine through the context menu and shows an error while using the raycast option.  

https://github.com/user-attachments/assets/a43cee94-7f5b-43e5-b8d9-0401ad5564d5

**Comment by EvanZhouDev** [comment]
Ah, well, it appears that this potentially might be 2 different issues:
@ozalisky In this case, it would be because Raycast Notes doesn't use the menubar at the top of the screen (as I have mentioned Edit > Writing Tools > Rewrite), which is what this extension automates. It doesn't go through the context menu, though that is a possibility I could explore.

**Comment by shreyshah33** [comment]
> Please check that you are able to access Apple Intelligence normally; Select the text, and click Edit (in your menubar at the top) > Writing Tools > Rewrite. This extension simply automates this process, so if it isn't available normally, the extension won't be able to automate it either.

Ah gotcha. Is there a potential feature where we can enable it in apps where writing tools are disabled via some form of temporary notes?

**Comment by EvanZhouDev** [comment]
Yes, in fact I've already experimented with this feature in my [apple-intelligence](https://github.com/EvanZhouDev/apple-intelligence) NPM library. Is this something you would be interested in?

The only downside I see to this is that it violates the Principle of Least Astonishment in the sense that you don't really know if it will directly act, or create a new note. Doing it in the note way takes significantly more time, and perhaps the user doesn't want that to occur.

Definitely open to hearing any suggestions regarding how to implement the UX though!

**Comment by rkulyniak** [comment]
I encountered the same issue and found that Raycast needs permission for `Automation - System Events`. 
Check if this helps in your case.

**Comment by AnirudhG07** [comment]
> I encountered the same issue and found that Raycast needs permission for `Automation - System Events`. Check if this helps in your case.

How do you give that permission. For me it only  shows up error like above shown. I don't find an option to toggle this. Pls tell how you did it.

**Comment by rkulyniak** [comment]
In the system settings.

<img width="702" alt="Screenshot 2024-12-22 at 20 00 02" src="https://github.com/user-attachments/assets/02d27391-ebfa-4bda-8839-ae33ae340b33" />

**Comment by AnirudhG07** [comment]
> In the system settings.

Mine is toggled for Raycast. 
Still unable to use...

**Comment by EvanZhouDev** [comment]
Wherever you are trying to use the extension, ensure that you can access Apple Intelligence by going to your Menubar > Edit > Writing Tools. This extension is unable to trigger Apple Intelligence without access through the Menubar.

**Comment by AnirudhG07** [comment]
Umm, If I select the text, right click, I see the "Writing Tools" option, but I dont see it in the menubar at the top in Edits. Do you know how I can enable it there? (Sorry if its a stupid question)

**Comment by EvanZhouDev** [comment]
Unfortunately, if it is not available through the menubar, there is not much that you can do about it. It depends on the app developer (or, for example, Raycast Notes is a floating window, so it doesn't have the menubar). In 95% of scenarios, it is available in the menubar, though.

**Comment by AnirudhG07** [comment]
I am using Safari, I dont see it there either. I dont think it will have that then. Sad... 
Is it not possible to use it with the right click options?? I am not familiar with whats available.

**Comment by EvanZhouDev** [comment]
I see it in Safari (and am able to use the extension fine):
<img width="1273" alt="image" src="https://github.com/user-attachments/assets/87091a0a-a331-478c-b876-8cf124fce492" />

Note that it also depends on the website. For example, Google Docs implements their own "textbox" which doesn't support Apple Intelligence.

**Comment by AnirudhG07** [comment]
Well then ill look into why safari is not giving this, maybe I need to update it or something......
